### ISSUE TITLE
🛠 Tooling: Build failures on main > contributors: "fetch is not set"

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#1227]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

From https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/7500363646/job/20418893198:

```plaintext
Run JoshuaKGoldberg/all-contributors-auto-action@v0.4.2
file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:10178
    throw new Error(
          ^

Error: fetch is not set. Please pass a fetch implementation as new Octokit({ request: { fetch }}). Learn more at https://github.com/octokit/octokit.js/#fetch-missing
    at fetchWrapper (file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:10178:11)
    at request2 (file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:10293:14)
    at hook (file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:4924:10)
    at async requestWithGraphqlErrorHandling (file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:9719:20)
    at async Job.doExecute (file:///home/runner/work/_actions/JoshuaKGoldberg/all-contributors-auto-action/v0.4.2/dist/index.js:117[5](https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/7500363646/job/20418893198#step:4:6)4:1[8](https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/7500363646/job/20418893198#step:4:9))
```

...is the Action not being run in Node >=18? I thought they all should be...

**Comment by JoshuaKGoldberg** [comment]
Huh. Definitely being run in Node >=18:

```plaintext
Run actions/setup-node@v4
  with:
    cache: pnpm
    node-version: 18
    always-auth: false
    check-latest: false
    token: ***
  env:
    PNPM_HOME: /home/runner/setup-pnpm/node_modules/.bin
Found in cache @ /opt/hostedtoolcache/node/18.19.0/x64
Environment details
  node: v18.19.0
  npm: 10.2.3
  yarn: 1.22.21
```

**Comment by JoshuaKGoldberg** [comment]
`all-contributors-for-repository` is on `octokit@3.0.0` right now: https://github.com/JoshuaKGoldberg/all-contributors-for-repository/blob/d24447e2f1ff4081f275b5d80da50f638c4cc2f8/pnpm-lock.yaml#L19. The latest is 3.1.2. I'll try upgrading packages to latest before filing an issue.

**Comment by JoshuaKGoldberg** [comment]
Well, that didn't fix it. Nor Node 20. Very confusing.

**Comment by JoshuaKGoldberg** [comment]
https://github.com/JoshuaKGoldberg/all-contributors-auto-action/blob/9ea3f66e6d0fc0615c5e508d279c28a136043c16/action.yml#L9

```plaintext
runs:
  main: dist/index.js
  using: node16
```

_Sigh_.

**Comment by JoshuaKGoldberg** [comment]
Fixed in 45f4b78 here.
